components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/HttpRequestOperation.java

```diff
@@ -321,11 +314,20 @@ public void onError(Throwable e) {
         public void onNext(ByteBuf chunk) {
             HttpObject msg = new DefaultHttpContent(chunk);
             channel.writeAndFlush(msg)
-                    .addListener(readNext)
-                    .addListener(readOrClose);
+                    .addListener((ChannelFuture future) -> {
+                        request(1);
```

Comment:
Should we encapsulate this in a function?

Reply:
I wouldn't bother. This class is going to get removed quite soon. 🤞